src/Fabulous.AST.Tests/TypeDefinitions/Enum.fs

```diff
@@ -0,0 +1,134 @@
+namespace Fabulous.AST.Tests.TypeDefinitions
+
+open Fabulous.AST.Tests
+open FSharp.Compiler.Text
+open Fantomas.Core.SyntaxOak
+open NUnit.Framework
+
+open Fabulous.AST
+
+open type Fabulous.AST.Ast
+
+module Enum =
+
+    [<Test>]
+    let ``Produces an enum`` () =
+        AnonymousModule() {
+            Enum("Colors") {
+                EnumCase("Red", "0")
+                EnumCase("Green", "1")
+                EnumCase("Blue", "2")
+            }
+        }
+
+        |> produces
+            """
+
+type Colors =
+    | Red = 0
+    | Green = 1
+    | Blue = 2
+"""
+
+    [<Test>]
+    let ``Produces an enum with SingleTextNode`` () =
+        AnonymousModule() {
+            Enum(SingleTextNode("Colors", Range.Zero)) {
+                EnumCase("Red", "0")
+                EnumCase("Green", "1")
+                EnumCase("Blue", "2")
+            }
+        }
+        |> produces
+            """
+
+type Colors =
+    | Red = 0
+    | Green = 1
+    | Blue = 2
+
+"""
+
+    [<Test>]
+    let ``Produces an enum case using EnumCaseNode`` () =
+        AnonymousModule() {
+            Enum("Colors") {
+                EnumCase("Red", "0")
+                EnumCase("Green", "1")
+                EnumCase("Blue", "2")
+
+                EnumCaseNode(
+                    None,
+                    None,
+                    None,
+                    SingleTextNode("Black", Range.Zero),
+                    SingleTextNode("=", Range.Zero),
+                    Expr.Constant(Constant.FromText(SingleTextNode("3", Range.Zero))),
+                    Range.Zero
+                )
+            }
+        }
+        |> produces
+            """
+
+type Colors =
+    | Red = 0
+    | Green = 1
+    | Blue = 2
+    | Black = 3
+
+"""
+
+    [<Test>]
+    let ``Produces an enum case using Widget and escape hatch`` () =
+        let enumCaseNode =
+            EnumCaseNode(
+                None,
+                None,
+                None,
+                SingleTextNode("Black", Range.Zero),
+                SingleTextNode("=", Range.Zero),
+                Expr.Constant(Constant.FromText(SingleTextNode("3", Range.Zero))),
+                Range.Zero
+            )
+
+        AnonymousModule() {
+            Enum("Colors") {
+                EnumCase("Red", "0")
+                EnumCase("Green", "1")
+                EnumCase("Blue", "2")
+                EscapeHatch(enumCaseNode)
```

Comment:
I really like this bit, that you need to wrap the `enumCaseNode` inside the `EscapeHatch`.
That makes it so clear in my opinion.